examples/ecs/state.rs

```diff
@@ -27,16 +29,44 @@ enum AppState {
 
 struct MenuData {
     button_entity: Entity,
+    continue_entity: Option<Entity>,
 }
 
+#[derive(Component)]
+struct RemoveWhenGameDone;
+
 const NORMAL_BUTTON: Color = Color::rgb(0.15, 0.15, 0.15);
 const HOVERED_BUTTON: Color = Color::rgb(0.25, 0.25, 0.25);
 const PRESSED_BUTTON: Color = Color::rgb(0.35, 0.75, 0.35);
 
-fn setup_menu(mut commands: Commands, asset_server: Res<AssetServer>) {
-    // ui camera
-    commands.spawn_bundle(UiCameraBundle::default());
-    let button_entity = commands
+fn setup_menu(
+    mut state: ResMut<State<AppState>>,
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+) {
+    let is_first_load = !state.inactives().contains(&AppState::InGame);
+    let button_entity = create_button(
+        &mut commands,
+        &*asset_server,
+        if is_first_load { "Play" } else { "Restart" },
+    );
+
+    let mut continue_entity = None;
+    if is_first_load {
+        // ui camera
+        commands.spawn_bundle(UiCameraBundle::default());
+    } else {
+        continue_entity = Some(create_button(&mut commands, &*asset_server, "Continue"));
+    }
+
+    commands.insert_resource(MenuData {
+        button_entity,
+        continue_entity,
+    });
+}
+
+fn create_button(commands: &mut Commands, asset_server: &AssetServer, title: &str) -> Entity {
```

Comment:
I like this; users often struggle to figure out how to write simple abstractions like this for manipulating entities.